Src/GeneratedTests/Discovery.Tests/DiscoveryTest.cs

```diff
@@ -0,0 +1,76 @@
+using Google.Apis.Http;
+using Google.Apis.Services;
+using NUnit.Framework;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.Threading;
+using System.Collections.Generic;
+using System.Text;
+using System.Reflection;
+using System.IO;
+
+namespace Google.Apis.Discovery.v1.Tests
+{
+    [TestFixture]
+    public class DiscoveryTest
+    {
+        class FakeHttpMessageHandler : ConfigurableMessageHandler
+        {
+            public FakeHttpMessageHandler(Dictionary<string, byte[]> data) : base(new HttpClientHandler())
+            {
+                _data = data;
+            }
+
+            private Dictionary<string, byte[]> _data;
+
+            protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                var result = new HttpResponseMessage(System.Net.HttpStatusCode.OK)
```

Comment:
Perhaps this could actually just fetch resources from the assembly, instead of being populated from a byte array?